### ISSUE TITLE
value controller reference changes on every render

### ISSUE THREAD

**Issue by msharifi99** [xnimorz/use-debounce#121]
**Describe the bug**
the reference of value controller object returned by `useDebounce`  keep changing on every render in consumer component. and it cause hooks that use the object as their dependencies to re-call.

**Expected behavior**
to reference does not change across re-renderes.

**use-debounce version: ^7.0.0**

**Comment by xnimorz** [comment]
Hey @msharifi99 
Thanks for the pull request, and I'm sorry for such a long response. 

I checked both issue and the pull request. 
Original solution suggested to check not the object which you receive, but the definite fieds:
```js
const [value, {cancel}] = useDebounce(123);
...
useEffect(() => {...}, [cancel]);
```

However, I agree that it may be difficult to support inside each component 

I just committed a fix for this: https://github.com/xnimorz/use-debounce/commit/ce543e30fd904653d5ca1d20c2f26e6347dee069

I don't use your PR because it increases the bundle size and this project target is to have a small bundle size and good developer experience at the same time. 

Thanks for reporting the issue!
